File: spring-boot-security-jwt-auth-mongodb-master/src/main/java/com/knf/dev/controllers/EmployeeController.java
package com.knf.dev.controllers;

import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.knf.dev.response.MessageResponse;

@CrossOrigin(origins = "*", maxAge = 4800)
@RestController
@RequestMapping("/api/test")
public class EmployeeController {

	@GetMapping("/all")
	public MessageResponse allAccess() {
		return new MessageResponse("Public ");
	}

	@GetMapping("/employee")
	@PreAuthorize("hasRole('EMPLOYEE') ")
	public MessageResponse employeeAccess() {

		return new MessageResponse("Employee zone");
	}

	@GetMapping("/admin")
	@PreAuthorize("hasRole('ADMIN')")
	public MessageResponse adminAccess() {
		return new MessageResponse("Admin zone");
	}
}

File: kotlin-spring-main/kotlin-springsecurity-jwt/src/main/kotlin/com/knf/dev/demo/kotlinspringsecurityjwt/controller/EmployeeController.kt
package com.knf.dev.demo.kotlinspringsecurityjwt.controller

import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.GetMapping
import com.knf.dev.demo.kotlinspringsecurityjwt.response.MessageResponse
import org.springframework.security.access.prepost.PreAuthorize

@CrossOrigin(origins = ["*"], maxAge = 4800)
@RestController
@RequestMapping("/api/test")
class EmployeeController {
    @GetMapping("/all")
    fun allAccess(): MessageResponse {
        return MessageResponse("Public ")
    }

    @GetMapping("/employee")
    @PreAuthorize("hasRole('EMPLOYEE') ")
    fun employeeAccess(): MessageResponse {
        return MessageResponse("Employee zone")
    }

    @GetMapping("/admin")
    @PreAuthorize("hasRole('ADMIN')")
    fun adminAccess(): MessageResponse {
        return MessageResponse("Admin zone")
    }
}